client/src/components/OfflineWarning/offline-warning.test.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { render, act, screen } from '@testing-library/react';
+
+import OfflineWarning from './offline-warning';
+
+jest.useFakeTimers();
+
+describe('<OfflineWarning />', () => {
+  it('renders null when isOnline, isServerOnline and isSignedIn are true', () => {
+    const { container } = render(
+      <OfflineWarning isOnline={true} isServerOnline={true} isSignedIn={true} />
+    );
+    act(() => jest.runOnlyPendingTimers());
```

Comment:
I think you're right about using runAllTimers instead of runOnlyPendingTimers since we are not dealing with recursive timers in the OfflineWarning component. Thank you for pointing this out.